applications/crossbar/src/crossbar_util.erl

```diff
@@ -1010,6 +1013,71 @@ format_emergency_caller_id_number(Context) ->
             end
     end.
 
+%% @public
+-spec maybe_refresh_fs_xml('user' | 'device', cb_context:context()) -> 'ok'.
+maybe_refresh_fs_xml(Kind, Context) ->
+    DbDoc = cb_context:fetch(Context, 'db_doc'),
+    Doc = cb_context:doc(Context),
+    Precondition =
+        (wh_json:get_value(<<"presence_id">>, DbDoc) =/=
+             wh_json:get_value(<<"presence_id">>, Doc))
+        or (wh_json:get_value([<<"media">>, <<"encryption">>, <<"enforce_security">>], DbDoc) =/=
+                wh_json:get_value([<<"media">>, <<"encryption">>, <<"enforce_security">>], Doc)),
+    maybe_refresh_fs_xml(Precondition, Kind, Context).
+
+-spec maybe_refresh_fs_xml(boolean(), 'user' | 'device', cb_context:context()) -> 'ok'.
+maybe_refresh_fs_xml('false', 'user', _Context) -> 'ok';
+maybe_refresh_fs_xml('true', 'user', Context) ->
+    Doc = cb_context:doc(Context),
+    AccountDb = cb_context:account_db(Context),
+    Realm     = wh_util:get_account_realm(AccountDb),
+    Id = wh_json:get_value(<<"_id">>, Doc),
+    Devices = get_devices_by_owner(AccountDb, Id),
+    lists:foreach(fun (DevDoc) -> refresh_fs_xml(Realm, DevDoc) end, Devices);
+maybe_refresh_fs_xml(Precondition, 'device', Context) ->
+    Doc   = cb_context:doc(Context),
+    DbDoc = cb_context:fetch(Context, 'db_doc'),
+    Realm = wh_util:get_account_realm(cb_context:account_db(Context)),
+    ( Precondition
+      or (sip(<<"username">>, DbDoc) =/= sip(<<"username">>, Doc))
+      or (sip(<<"password">>, DbDoc) =/= sip(<<"password">>, Doc))
+      or (wh_json:get_value(<<"owner_id">>, DbDoc) =/=
+              wh_json:get_value(<<"owner_id">>, Doc))
+    ) andalso
+        refresh_fs_xml(Realm, DbDoc),
+    'ok'.
+
+-spec refresh_fs_xml(ne_binary(), wh_json:object()) -> 'ok'.
+refresh_fs_xml(Realm, Doc) ->
+    case sip(<<"username">>, Doc) of
+        'undefined' -> 'ok';
+        Username ->
+            lager:debug("flushing fs xml for user '~s' at '~s'", [Username,Realm]),
+            Req = [{<<"Username">>, Username}
+                   ,{<<"Realm">>, Realm}
+                   | wh_api:default_headers(?APP_NAME, ?APP_VERSION)
+                  ],
+            wh_amqp_worker:cast(Req, fun wapi_switch:publish_fs_xml_flush/1)
+    end.
+
+-spec sip(ne_binary(), wh_json:object()) -> api_binary().
+sip(Key, JObj) ->
```

Comment:
Not a very descriptive name for a function. Seems like it should be `get_sip_attribute` or something along those lines.